src/rgblob/core.cljs

```diff
@@ -0,0 +1,130 @@
+(ns rgblob.core
+  (:require [reagent.core :as reagent :refer [atom]]
+      [goog.events :as events]
+      [rgblob.game :as game]
+      [rgblob.sprite :as rs]
+      [rgblob.player :as player]
+      [rgblob.util :as util :refer [log log-obj]])
+  (:import [goog.events EventType]))
+
+(enable-console-print!)
+
+(def jquery (js* "$"))
+
+(def code->input
+  "Map from keycode to user intention.
+  
+  Borrowed from https://gist.github.com/tomconnors/8460406"
+  {13 :enter
+   27 :escape
+   37 :left
+   38 :up
+   39 :right
+   40 :down
+   65 :left
+   87 :up
+   68 :right
+   83 :down
+   })
+
+(defonce state 
+    (atom {:last-key :none}))
+
+(defn assoc-in-multiple [base & settings-pairs]
+  (reduce (fn set-settings-pair [acc pair]
+            (assoc-in acc [(first pair)] (second pair)))
+          base
+          settings-pairs))
+
+(defn default-square-props [i j]
+  {:width 1
+   :height 1
+   :fill "#AAA"
+   :x i
+   :y j
+   :stroke "#777"})
+
+(defn render-square 
+  "Renders a square of a given type, :red, :green, :blue, or :white"
+  [x-pos y-pos color]
+  [:rect (assoc-in-multiple (default-square-props x-pos y-pos)
+                             [:fill ])])
+
+
+(defn rgblob []
+    [:div { :class-name "container" }
+      [:h1 "RGBlob"]
+      [:canvas {:width 512
+                :height 512
+                :id "game"}]
+      ])
+
+
+(reagent/render-component [rgblob]
+                          (. js/document (getElementById "app")))
+
+(def KEY-EVENT "keydown")
+
+(def canvas (js/document.getElementById "game"))
+
+(defn handle-key-down [e]
+  (let [input (code->input (.-keyCode e))]
+    (if (not (nil? input)) (do 
+      (game/input input)
+      (.preventDefault e)))))
+
+(def CANVAS (.getElementById js/document "game"))
+
+(defn get-context []
+  (.getContext CANVAS "2d"))
+
+(defn get-blob []
```

Comment:
You're not getting something here, you're creating something, and furthermore, it's a different something every time.  So this is really `make-blob` or `create-blob`